lambda/lambda.ml

```diff
@@ -756,6 +785,13 @@ let rename idmap lam =
   let s = Ident.Map.map (fun new_id -> Lvar new_id) idmap in
   subst update_env s lam
 
+let duplicate lam =
+  subst
+    (fun _ _ env -> env)
```

Comment:
Are you sure that `duplicate`  should have a neutral `update_env` (first) argument ?

Reply:
There is nothing that feels right that we can do here, because at this point we don't have access to the bound-name-renaming map. (The `update_env` interface is a bit akward.) On the other hand, I could use the bound-name-renaming map at the point in the function where `update_env` is called, to duplicate the bindings for the fresh bound variables. That sounds like a correct thing to do (modulo the fact that we should in fact rename the bindings, but we don't have the interface to do it), thanks for the suggestion.

Reply:
Hm apparently I had forgotten about it, but the code I wrote already does this (see the fold on `l` in the `Levent` case,)